### PR TITLE
CVSL-218 Check your answers, view, print and approve a licence.

### PR DESCRIPTION
**Pull request by timharrison-moj**
* Lots of content changes (titles, condition counters, change links and wording).
* Expanded condition text with in-line values shown on all summary screens.
* Removed rejection link (hidden) - but integration test forces its use.
* Updated unit and integration tests for all content changes.
* Use of short-names for some categories on summary pages (replaces full category, if present, and written to DB via API)
* Adds missing probation regions, and appends `..Probation Region`" to all e.g. `London Probation Region` - so condition sentences don't need to be changed.
* Simplified tests by defining and re-using a single licence object (using spread syntax to change type/status).

### CODE REVIEW

File: integration_tests/mockApis/licence.ts [hunk 1]
```diff
@@ -120,7 +120,7 @@ export default {
             {
               id: 3,
               code: 'a7c57e4e-30fe-4797-9fe7-70a35dbd7b65',
-              category: 'Participation in, or co-operation with, a programme or set of activities',
+              category: 'Programmes or activities',
               sequence: 2,
```
Comment (timharrison-moj) [inline integration_tests/mockApis/licence.ts:124]: For some categories, this is what will be supplied to the API when updating them, so it correct that the mock API should return the shortened names.

File: integration_tests/pages/approvalView.ts [hunk 2]
```diff
@@ -17,7 +17,8 @@ export default class ApprovalViewPage extends Page {
   }
 
   clickReject = (): ConfirmRejectPage => {
-    cy.get(this.rejectLicenceButtonId).click()
+    // Force: true will click even if a hidden element
+    cy.get(this.rejectLicenceButtonId).click({ force: true })
     return Page.verifyOnPage(ConfirmRejectPage)
```
Comment (timharrison-moj) [inline integration_tests/pages/approvalView.ts:22]: The reject link has been hidden by request.

File: server/routes/approvingLicences/handlers/approvalView.ts [hunk 3]
```diff
@@ -10,7 +11,9 @@ export default class ApprovalViewRoutes {
 
     // Check whether this licence is still in a SUBMITTED state - back button pressed - avoid re-approval
     if (licence?.statusCode === LicenceStatus.SUBMITTED) {
-      res.render('pages/approve/view')
+      const expandedLicenceConditions = expandAdditionalConditions(licence.additionalLicenceConditions)
+      const expandedPssConditions = expandAdditionalConditions(licence.additionalPssConditions)
+      res.render('pages/approve/view', { expandedLicenceConditions, expandedPssConditions })
     } else {
       res.redirect(`/licence/approve/cases`)
     }
```
Comment (timharrison-moj) [inline server/routes/approvingLicences/handlers/approvalView.ts:19]: Expands the conditions for in-line text and data, and passes through to the views.

File: server/services/licenceService.ts [hunk 4]
```diff
@@ -159,7 +159,9 @@ export default class LicenceService {
           return {
             code: conditionCode,
             sequence: index,
-            category: getAdditionalConditionByCode(conditionCode)?.category,
+            category:
+              getAdditionalConditionByCode(conditionCode)?.categoryShort ||
+              getAdditionalConditionByCode(conditionCode)?.category,
             text: getAdditionalConditionByCode(conditionCode)?.text,
           }
```
Comment (timharrison-moj) [inline server/services/licenceService.ts:166]: For conditions which define  a shorter name, this is what we send to the API to store. Its only used on summary screens, and these all use the short description anyway. The conditions select page gets the descriptions from config so always uses the long description.

File: server/config/conditions.ts [hunk 5]
```diff
@@ -110,8 +110,8 @@ export default {
       {
         code: '5db26ab3-9b6f-4bee-b2aa-53aa3f3be7dd',
         category: 'Residence at a specific place',
-        text: 'You must reside within the [INSERT REGION] probation region while of no fixed abode, unless otherwise approved by your supervising officer.',
-        tpl: 'You must reside within the {probationRegion} probation region while of no fixed abode, unless otherwise approved by your supervising officer.',
```
Comment (smcveigh941) [inline server/config/conditions.ts:114]: Whats the difference in putting probation region here or on the end of each option? Will we have some which dont have probation region on the end?

File: server/views/pages/approve/view.njk [hunk 6]
```diff
@@ -13,7 +13,15 @@
...
         <div class="govuk-grid-column-three-quarters">
-            <h1 class="govuk-heading-l">Approve a licence for {{ licence.forename }} {{ licence.surname }}</h1>
+            {% if licence.typeCode === 'AP' %}
+                <h1 class="govuk-heading-l">Approve licence for {{ licence.forename }} {{ licence.surname }}</h1>
+            {% elseif licence.typeCode === 'AP_PSS' %}
+                <h1 class="govuk-heading-l">Approve licence and post sentence supervision order for {{ licence.forename }} {{ licence.surname }}</h1>
+            {% elseif licence.typeCode === 'PSS' %}
+                <h1 class="govuk-heading-l">Approve post sentence supervision order for {{ licence.forename }} {{ licence.surname }}</h1>
+            {% else %}
+                <h1 class="govuk-heading-l">Approve licence type {{ licence.typeCode }} for {{ licence.forename }} {{ licence.surname }}</h1>
```
Comment (smcveigh941) [inline server/views/pages/approve/view.njk:23]: Is this possible?

File: server/views/pages/create/checkAnswers.njk [hunk 7]
```diff
@@ -14,8 +14,15 @@
...
-            <span class="govuk-caption-m">{{ licence.forename + " " + licence.surname }}</span>
-            <h1 class="govuk-heading-l">Check your answers</h1>
+            {% if licence.typeCode === 'AP' %}
+                <h1 class="govuk-heading-l">Check licence details for {{ licence.forename }} {{ licence.surname }}</h1>
+            {% elseif licence.typeCode === 'AP_PSS' %}
+                <h1 class="govuk-heading-l">Check licence and post sentence supervision order details for {{ licence.forename }} {{ licence.surname }}</h1>
+            {% elseif licence.typeCode === 'PSS' %}
+                <h1 class="govuk-heading-l">Check post sentence supervision order details for {{ licence.forename }} {{ licence.surname }}</h1>
+            {% else %}
+                <h1 class="govuk-heading-l">Check licence type {{ licence.typeCode }} for {{ licence.forename }} {{ licence.surname }}</h1>
```
Comment (smcveigh941) [inline server/views/pages/create/checkAnswers.njk:24]: Same question here

**Review by smcveigh941 — APPROVED** [review]
LGTM 👏🏼

File: server/config/conditions.ts (hunk 5 above)
Comment (timharrison-moj) [reply server/config/conditions.ts:114]: The original condition text didn't have it in so we added it into the template text. This makes it part of the select list value again - but this area is, I know, being looked at again by Ricky - talk of a radio here, so can be revisited.

File: server/views/pages/approve/view.njk (hunk 6 above)
Comment (timharrison-moj) [reply server/views/pages/approve/view.njk:23]: Not at present. Fingers crossed - unless its gone wrong.

**Pull request merged**